### PR TITLE
#672 Add documentation to the dist.

### LINKED ISSUE #672: Restore documentation to source manifest
Please restore the documentation source, to the source manifest (so that it is included in the Python sdist).

In commit 6463b5a8601957a40ee83aaf12fcad9e5a25d458 a `recursive-exclude` was added to the 'Manifest.in' file, deliberately excluding the documentation source from the sdist.

That commit has no explanation for the change; the commit message in its entirety is "fixes".

Please revert that change, and include the documentation in the sdist as before.

### PR DESCRIPTION
**Pull request by adiroiban**
# Description

Fixes #672

Update the pyproject.toml

# How to test

Run `python -m build`.
It will generate  `.tar.gz` and `.whl` files inside the `dist/` folder.
Check that the documentation source files are included in the `.tar.gz` version.
Check that the documentation files are not included in the `.whl` version.

`.whl` is just a zip file.

### CODE REVIEW

**Comment by adiroiban** [comment]
This is ready for review

needs-review

**Review by twm — APPROVED** [review]
LGTM. I do notice that there are some files in the .tar.gz that probably should be omitted:

```
towncrier-24.8.0.dev0/.github/CODEOWNERS
towncrier-24.8.0.dev0/.github/PULL_REQUEST_TEMPLATE.md
towncrier-24.8.0.dev0/.github/workflows/ci.yml
towncrier-24.8.0.dev0/.git-blame-ignore-revs
```

Perhaps also these, as I can't imagine that pre-commit is run by packagers:

```
towncrier-24.8.0.dev0/.pre-commit-config.yaml
towncrier-24.8.0.dev0/.pre-commit-hooks.yaml
```

None of that blocking, though. Feel free to merge. Thanks!

**Comment by adiroiban** [comment]
Many thanks for the review.

I will exclude github and pre-commit files.

> Perhaps also these, as I can't imagine that pre-commit is run by packagers:

I think that the idea of source dist, is not necesarily to be used by the packages, but to allow a person to do software development ... without `git clone`

I have also removed the pre-commit config.

The developemnt of `towncrier` should be done using `git clone` and pre-commit doesn't make much sense without git.

File: pyproject.toml [hunk 1]
```diff
@@ -63,7 +63,10 @@ Distribution = "https://pypi.org/project/towncrier"
 exclude = [
     "admin",
     "bin",
-    "docs",
+    ".github",
+    ".git-blame-ignore-revs",
+    ".pre-commit-config.yaml,
```
Comment (adiroiban) [inline pyproject.toml]: fix toml list syntax

```suggestion
    ".pre-commit-config.yaml",
```

**Pull request merged**